### PR TITLE
Allow configuration ActiveJob queue for auto_index

### PR DESCRIPTION
**Pull request by bagedevimo**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | no, provided #438 merged first / removed  
| Related Issue     | Makes progress towards https://github.com/algolia/algoliasearch-rails/issues/85
| Need Doc update   | yes

## Describe your change

Copy-pasted from commit message:
```
    Allow library users to specify default ActiveJob queue for indexing

    The current auto_indexing behaviours are great, but force a user to
    effectively reimplement the default enqueue behaviour if they just wanna
    use a different ActiveJob queue. Instead, allow the queue_name to
    specified as a configuraiton option, but fallback to the default if no
    value is set.

    Someone who just updates to a version with this commit should see no
    difference until they supply a value for the configuration, if they
    wish.
```

### CODE REVIEW

**Comment by bagedevimo** [comment]
It's worth resolving if #438 is the direction Algolia would like to go before making decisions / reviewing this. After that PR is changed / merged / closed, i'll adjust this PR accordingly.

**Comment by DevinCodes** [comment]
Thank you for the PR @bagedevimo ! Sorry in advance but I won't have time to review this until after next week, but wanted to let you know I've seen it and will get to it. Apologies for the delay!

**Comment by DevinCodes** [comment]
Thank you for this change @bagedevimo ! Would like to merge this in if the comments in the other PR are addressed / updated in this PR 🙂

**Comment by bagedevimo** [comment]
Hi @DevinCodes i've updated this MR to use the stuff from #438

File: spec/integration_spec.rb
Comment (DevinCodes) [inline spec/integration_spec.rb:591]: Why do we need to add this line here?

**Review by DevinCodes — CHANGES_REQUESTED** [review]
Thank you for the update! Overall the PR looks good to me, I just have one question left 🙂

File: spec/integration_spec.rb
Comment (bagedevimo) [reply spec/integration_spec.rb:591]: ah, yes! i forgot to mention this - copied from the commit message:

```
Specify the GlobalID app name
Required for GlobalID to generate names, but wasn't previously required
as no objects were actually being enqueued and therefore not actually
generating names.

In a rails environment, this would be automatically configured by the
railtie in GlobalID [1].

https://github.com/rails/globalid/blob/b0f5af88e1baf966a4309f97321bf71dcfc6f248/lib/global_id/railtie.rb#L20
```

**Comment by bagedevimo** [comment]
It's not clear to me what the spec failures are - I can't reproduce the utilities spec failure locally, and the others seem to be failing due to me not having permissions to get a key?

**Review by DevinCodes — APPROVED** [review]
Hey @bagedevimo ,

no worries about the CI: we're aware it's flaky, especially for external contributors. I've started a run of the CI that should have the proper permissions. 

Regarding the failing utility test, it's been failing for a while, it's not related to this PR. It's on my todo list to fix this test but haven't had the time yet.

The PR looks good to me, so happy to merge and release this. Given that Black Friday is coming up next week (historically a busy period for a lot of the users of this gem), I'd like to hold off on releasing this until the week after Black Friday. I'll let you know when this is released!

**Comment by bagedevimo** [comment]
I raised https://github.com/algolia/algoliasearch-rails/issues/440 to cover some brief investigation into the likely cause of some of those specs. Feel free to just close it, but thought it might be helpful to briefly document what I'd discovered.

**Comment by DevinCodes** [comment]
@bagedevimo thanks again, we just released this in v2.3.1 of the gem! https://rubygems.org/gems/algoliasearch-rails/versions/2.3.1

**Pull request merged**
